### PR TITLE
Fix pbr world normal mismatch

### PR DESCRIPTION
**Pull request by deltakosh**

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/15430/merge

Test environment: 
https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15430/merge/index.html

To test a playground add it to the URL, for example:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15430/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/15430/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/15430/merge
https://gui.babylonjs.com/?snapshot=refs/pull/15430/merge
https://nme.babylonjs.com/?snapshot=refs/pull/15430/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/15430/merge#BCU1XR#0

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15430/merge/testResults/webgl2playwright/index.html

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15430/merge/testResults/webgpuplaywright/index.html

**Review by sebavan — APPROVED** [review]

File: packages/dev/core/src/Materials/Node/Blocks/PBR/pbrMetallicRoughnessBlock.ts [hunk 1]
```diff
@@ -1068,11 +1074,16 @@ export class PBRMetallicRoughnessBlock extends NodeMaterialBlock {
...
 
         let worldPosVarName = this.worldPosition.associatedVariableName;
+        let worldNormalVarName = this.worldNormal.associatedVariableName;
         if (this.generateOnlyFragmentCode) {
             worldPosVarName = state._getFreeVariableName("globalWorldPos");
             state._emitFunction("pbr_globalworldpos", `vec3${state.fSuffix} ${worldPosVarName};\n`, comments);
             state.compilationString += `${worldPosVarName} = ${this.worldPosition.associatedVariableName}.xyz;\n`;
 
+            worldNormalVarName = state._getFreeVariableName("globalWorldNormal");
+            state._emitFunction("pbr_globalworldnorm", `vec3${state.fSuffix} ${worldNormalVarName};\n`, comments);
```
Comment (Popov72) [inline packages/dev/core/src/Materials/Node/Blocks/PBR/pbrMetallicRoughnessBlock.ts:1084]: Do we need this line? I don't see "pbr_globalworldnorm" used anywhere in the code?

Note: it's the same for "pbr_globalworldpos".

Comment (deltakosh) [reply packages/dev/core/src/Materials/Node/Blocks/PBR/pbrMetallicRoughnessBlock.ts:1084]: yeah this is over protective I agree. I'll remove that

**Pull request merged**